src/Core/Label.js

```diff
@@ -97,14 +96,16 @@ class Label extends THREE.Object3D {
                 if (style.text.haloWidth > 0) {
                     this.content.classList.add('itowns-stroke-single');
                 }
-                style.applyToHTML(this.content, sprites);
+                style.applyToHTML(this.content)
+                    .then(() => {
+                        this.icon = this.content.getElementsByClassName('itowns-icon')[0];
```

Comment:
What if i add a new event 'loaded' that will be send when the icon is loaded ?
 and then catch the event in Label.js
 ```
                style.applyToHTML(this.content)
                    .then((icon) => {
                        if (icon) {
                            icon.addEventListener('loaded', ()  => {
                                this.icon = this.content.getElementsByClassName('itowns-icon')[0];
                            });
                        }
                    });
```

Reply:
I prefer using await or a promise that resolves with `cIcon` in `applyToHMTL` because here you rely on an event that you manually add and on its hardcoded name in two different places. Similarly, I think we should not rely on the `itowns-icon` class name to retrieve the icon. Resolving / returning `cIcon` in `applyToHTML` would resolve both these issues.

Reply:
I think, I did find a way to do what you propose: see 502ad620fce5a76b219dc2678b8398a806dc835b.
What do you think @jailln ?

Reply:
I did so test and it seems that the test icon.complete is indeed not necessary.
I propose in the last commit some modification inspired by your code. I put out the addicon function from applyToHTML to prevent it being created many time and I fell like it makes the code a bit more readable (and from my point of view a bit more similare to applyToCanvasPolygon), WDYT ?

Reply:
ok :+1: can you just update the doc of `applyToHTML` (especially the @returns please?)

Reply:
done